wdl2cwl/main.py

```diff
@@ -400,8 +405,8 @@ def get_expr_apply(self, wdl_apply_expr: WDL.Expr.Apply) -> str:  # type: ignore
             right_operand = self.get_expr(right_operand)  # type: ignore
             left_operand_value = self.get_expr(left_operand)
             if getattr(left_operand, "function_name", None) == "basename":
-                treat_as_optional = True
                 referer = wdl_apply_expr.parent.name  # type: ignore
+                treat_as_optional = True if referer in self.non_static_values else False
```

Comment:
This ensures that the variable expression calling the basename function is declared within the input as optional and not a static variable.